modelexpress_client/python/modelexpress/shape_descriptors.py

```
@@ -0,0 +1,374 @@
+        #     ``.to_local``) has shape == local-view by construction.
+        # Compute global vs local accordingly.
+        try:
+            from torch.distributed.tensor import DTensor as _RealDTensor
+        except ImportError:  # pragma: no cover — handled at module import
+            _RealDTensor = None  # type: ignore[assignment]
+
+        if _RealDTensor is not None and isinstance(tensor, _RealDTensor):
+            global_shape = tuple(int(s) for s in tensor.shape)
+            try:
+                local_extent = int(tensor.to_local().shape[p.dim])
+            except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)